### PR TITLE
[Config] Fix Set list path

### PR DESCRIPTION
**Pull request by samsonasik**
|    Q          |   A
|-------------- | ------
| Bugfix        | yes

### CODE REVIEW

**Comment by samsonasik** [comment]
Hi @TomasVotruba,

I tried create custom Rector `SetList` class for laminas-servicemanager-migration, named: `Laminas\ServiceManager\Migration\Rector\Set\ValueObject\SetList`, so it need to be usable to:

```php
<?php
// rector.php
declare(strict_types=1);

use Rector\Core\Configuration\Option;
use Laminas\ServiceManager\Migration\Rector\Set\ValueObject\SetList;
use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $containerConfigurator): void {
    $parameters = $containerConfigurator->parameters();
    $parameters->set(Option::PATHS, [__DIR__ . '/module']);

    $parameters->set(Option::SETS, [
        SetList::LAMINAS_SERVICEMANGER_40,
    ]);
};
```

However, when I tried, it shows that `Set` was not found:

```bash
➜  laminas-mvc-skeleton vendor/bin/rector process module/Application/src/Listener                                                                                     

                                                                                                                        
 [ERROR] Set                                                                                                            
         "/Users/samsonasik/www/laminas-mvc-skeleton/vendor/laminas/laminas-servicemanager-migration/src/Rector/Set/Valu
         eObject/../../../../config/rector/set/laminas-servicemanager-40.php" was not found                             
                                                                                                                        

 ! [NOTE] Pick one of:                                                                                                  

 * defluent
 * action-injection-to-constructor-injection
 * array-str-functions-to-static-call
 * cakephp30
 * cakephp34
 .
 .
 .
```

I re-check the path even with realpath, and the path actually correct. 

Do you have idea how to make it works? Thank you.

**Comment by TomasVotruba** [comment]
Hi, I'd try to run the command with `--debug`. See the class this exception is thrown.

**Comment by samsonasik** [comment]
@TomasVotruba thank you, using `--debug` got the same output. Also tried `dev-main as 0.9.31` to get latest got different error, but the `Pick one of` is now exploded by `-`:

```bash
vendor/bin/rector process module/Application/src/Listener --debug

                                                                                                                        
 [ERROR] Set                                                                                                            
         "/Users/samsonasik/www/laminas-mvc-skeleton/vendor/laminas/laminas-servicemanager-migration/src/Rector/Set/Valu
         eObject/../../../../config/rector/set/laminas-servicemanager-40.php" was not found                             
                                                                                                                        

 ! [NOTE] Pick one of:                                                                                                  

 * d-e-f-l-u-e-n-t
 * a-c-t-i-o-n-i-n-j-e-c-t-i-o-n-t-o-c-o-n-s-t-r-u-c-t-o-r-i-n-j-e-c-t-i-o-n
 * a-r-r-a-y-s-t-r-f-u-n-c-t-i-o-n-s-t-o-s-t-a-t-i-c-c-a-l-l
 * c-a-k-e-p-h-p30
 * c-a-k-e-p-h-p34
 * c-a-k-e-p-h-p35
 * c-a-k-e-p-h-p36
 * c-a-k-e-p-h-p37
 * c-a-k-e-p-h-p38
 * c-a-k-e-p-h-p40
 * c-a-k-e-p-h-p41
 * c-a-k-e-p-h-p42
 * c-a-k-e-p-h-p-f-l-u-e-n-t-o-p-t-i-o-n-s
 * c-o-d-e-q-u-a-l-i-t-y
 ```
 
 The realpath of set actually found:
 
 ```bash
 php > echo realpath('/Users/samsonasik/www/laminas-mvc-skeleton/vendor/laminas/laminas-servicemanager-migration/src/Rector/Set/ValueObject/../../../../config/rector/set/laminas-servicemanager-40.php');
 
/Users/samsonasik/www/laminas-mvc-skeleton/vendor/laminas/laminas-servicemanager-migration/config/rector/set/laminas-servicemanager-40.php
```

**Comment by samsonasik** [comment]
@TomasVotruba it seems it because the rector's `RectorSetProvider::SET_LIST_CLASSES` defined constant so it can't be customized, ref:

https://github.com/rectorphp/rector/blob/c815334bed781c555f08dff82ce7861d758e9e7a/packages/Set/RectorSetProvider.php#L30

I will check if I can update it to make it extendable.

**Comment by samsonasik** [comment]
I've added handling use custom set list at https://github.com/rectorphp/rector/pull/5894 🎉

**Comment by samsonasik** [comment]
I got the solution as commented at https://github.com/rectorphp/rector/issues/5891#issuecomment-803291383 . I will update the documentation.

**Comment by samsonasik** [comment]
tested and it works 🎉 

Screen Shot 2021-03-20 at 18 04 03

**Comment by samsonasik** [comment]
Documentation config example updated. Merging ... 🎉

File: src/Rector/Set/ValueObject/SetList.php [hunk 1]
```diff
@@ -4,5 +4,5 @@
 
 final class SetList
 {
-    public const LAMINAS_SERVICEMANGER_40 = __DIR__ . '/../../../config/set/laminas-servicemanager-40.php';
-}
\ No newline at end of file
+    public const LAMINAS_SERVICEMANGER_40 = __DIR__ . '/../../../../config/rector/set/laminas-servicemanager-40.php';
```
Comment (TomasVotruba) [inline src/Rector/Set/ValueObject/SetList.php:7]: Btw, bugs like wrong paths like these can be discovered by PHPStan

See https://github.com/symplify/phpstan-rules/blob/main/docs/rules_overview.md#nomissingdirpathrule

Comment (samsonasik) [reply src/Rector/Set/ValueObject/SetList.php:7]: @TomasVotruba thank you 👍

**Pull request merged**